WordPress/Classes/Services/PostService.swift

```diff
@@ -0,0 +1,678 @@
+                                            return
+                                        }
+
+                                        let post = self.find(byID: postID, blog: blogInContext) ?? self.makePost(for: blogInContext)
+
+                                        if let post = post {
+                                            self.update(post, with:remotePost)
+                                        }
+                                        ContextManager.sharedInstance().saveContext(self.managedObjectContext)
+
+                                        success?(post)
+                                    }
+                },
+                                 failure: { (error: NSError?) in
+                                    if let failure = failure {
+                                        self.managedObjectContext.performBlock {
+                                            failure(error)
+                                        }
+                                    }
+                }
+            )
+        }
+    }
+
+    ///  Sync an initial batch of posts from the specified blog
+    ///  - note: Please note that success and/or failure are called in the context of the
+    ///  NSManagedObjectContext supplied when the PostService was initialized, and may not
+    ///  run on the main thread.
+    ///
+    ///  - parameter type:    The type (post or page) of post to sync
+    ///  - parameter blog:    The blog that has the posts
+    ///  - parameter success: A success block
+    ///  - parameter failure: A failure block
+    func sync(type: PostType, blog: Blog, success: SuccessPostsBlock, failure: FailureBlock) {
+        sync(type, blog: blog, options: nil, success: success, failure: failure)
+    }
+
+    ///  Sync an initial batch of posts from the specified blog
+    ///  - note: Please note that success and/or failure are called in the context of the
+    ///  NSManagedObjectContext supplied when the PostService was initialized, and may not
+    ///  run on the main thread.
+    ///
+    ///  - parameter type:    The type (post or page) of post to sync
+    ///  - parameter blog:    The blog that has the posts
+    ///  - parameter options: Sync options for specific request parameters
+    ///  - parameter success: A success block
+    ///  - parameter failure: A failure block
+    func sync(type: PostType, blog: Blog, options: PostServiceSyncOptions?, success: SuccessPostsBlock?, failure: FailureBlock?) {
+        if let remote = remote(for:blog) {
+            let blogID = blog.objectID
+            let remoteOptions = remoteSyncParameters(for:remote, with:options)
+            remote.getPostsOfType(type.rawValue, options: remoteOptions, success: { (remotePosts) in
+                    self.managedObjectContext.performBlock() {
+                        do {
+                            if let blogInContext = try self.managedObjectContext.existingObjectWithID(blogID) as? Blog {
```

Comment:
This also may not handle the similar case in which `blogInContext` was `nil` but should still log the error.